From cf2478432291556e3045e9271708cf6fb8015d49 Mon Sep 17 00:00:00 2001
From: "song.yu" <song.yu@sophgo.com>
Date: Thu, 19 Mar 2026 18:02:17 +0800
Subject: [PATCH] [fix](vo):Adapt the I80 panel on alios

[description]:NA
[root cause]:NA
[JIRA ID]:CV184X-1121
[chip project]:CV184X
[side effects]:NA
[Affected *.ko/*.so/*.a] :NA

Change-Id: Ia567369931ac603a54bc2dbc90574a11b383489c
---
 components/cvi_mmf_sdk/cvi_panel/cvi_panels.h | 139 +++++++++++++++++
 components/cvi_mmf_sdk/cvi_panel/dsi_panels.h |  93 -----------
 .../cvi_mmf_sdk/cvi_panel/hw_mcu_st7789v3.h   | 124 +++++++++++++++
 .../cvi_platform/media/src/media_video.c      | 145 +++++++++++-------
 4 files changed, 350 insertions(+), 151 deletions(-)
 create mode 100644 components/cvi_mmf_sdk/cvi_panel/cvi_panels.h
 delete mode 100644 components/cvi_mmf_sdk/cvi_panel/dsi_panels.h
 create mode 100644 components/cvi_mmf_sdk/cvi_panel/hw_mcu_st7789v3.h

diff --git a/components/cvi_mmf_sdk/cvi_panel/cvi_panels.h b/components/cvi_mmf_sdk/cvi_panel/cvi_panels.h
new file mode 100644
index 00000000..fb2a3725
--- /dev/null
+++ b/components/cvi_mmf_sdk/cvi_panel/cvi_panels.h
@@ -0,0 +1,139 @@
+/*
+ * Copyright (C) Cvitek Co., Ltd. 2019-2021. All rights reserved.
+ *
+ * File Name: cvi_panels.h
+ * Description:
+ */
+
+#ifndef __CVI_PANEL_H__
+#define __CVI_PANEL_H__
+
+#include <cvi_comm_vo.h>
+
+struct panel_desc_s {
+  char *panel_name;
+  struct combo_dev_cfg_s *dev_cfg;
+  const struct hs_settle_s *hs_timing_cfg;
+  const struct dsc_instr *dsi_init_cmds;
+  int dsi_init_cmds_size;
+  const VO_PUB_ATTR_S stVoPubAttr;
+  const VO_LVDS_ATTR_S *LvdsAttr;
+  const VO_BT_ATTR_S *BtAttr;
+};
+
+#if CONFIG_PANEL_HX8394
+#include "dsi_hx8394_evb.h"
+static struct panel_desc_s panel_desc = {
+    .panel_name = "HX8394-720x1280",
+    .dev_cfg = &dev_cfg_hx8394_720x1280,
+    .hs_timing_cfg = &hs_timing_cfg_hx8394_720x1280,
+    .dsi_init_cmds = dsi_init_cmds_hx8394_720x1280,
+    .dsi_init_cmds_size = ARRAY_SIZE(dsi_init_cmds_hx8394_720x1280)};
+#elif CONFIG_PANEL_ILI9881C
+#include "dsi_ili9881c.h"
+static struct panel_desc_s panel_desc = {
+    .panel_name = "ILI9881C-720x1280",
+    .dev_cfg = &dev_cfg_ili9881c_720x1280,
+    .hs_timing_cfg = &hs_timing_cfg_ili9881c_720x1280,
+    .dsi_init_cmds = dsi_init_cmds_ili9881c_720x1280,
+    .dsi_init_cmds_size = ARRAY_SIZE(dsi_init_cmds_ili9881c_720x1280)};
+#elif CONFIG_PANEL_ILI9881D
+#include "dsi_ili9881d.h"
+static struct panel_desc_s panel_desc = {
+    .panel_name = "ILI9881D-720x1280",
+    .dev_cfg = &dev_cfg_ili9881d_720x1280,
+    .hs_timing_cfg = &hs_timing_cfg_ili9881d_720x1280,
+    .dsi_init_cmds = dsi_init_cmds_ili9881d_720x1280,
+    .dsi_init_cmds_size = ARRAY_SIZE(dsi_init_cmds_ili9881d_720x1280)};
+#elif CONFIG_PANEL_JD9366AB
+#include "dsi_jd9366ab.h"
+static struct panel_desc_s panel_desc = {
+    .panel_name = "JD9366AB-800x1280",
+    .dev_cfg = &dev_cfg_jd9366ab_800x1280,
+    .hs_timing_cfg = &hs_timing_cfg_jd9366ab_800x1280,
+    .dsi_init_cmds = dsi_init_cmds_jd9366ab_800x1280,
+    .dsi_init_cmds_size = ARRAY_SIZE(dsi_init_cmds_jd9366ab_800x1280)};
+#elif CONFIG_PANEL_NT35521
+#include "dsi_nt35521.h"
+static struct panel_desc_s panel_desc = {
+    .panel_name = "NT35521-800x1280",
+    .dev_cfg = &dev_cfg_nt35521_800x1280,
+    .hs_timing_cfg = &hs_timing_cfg_nt35521_800x1280,
+    .dsi_init_cmds = dsi_init_cmds_nt35521_800x1280,
+    .dsi_init_cmds_size = ARRAY_SIZE(dsi_init_cmds_nt35521_800x1280)};
+#elif CONFIG_PANEL_OTA7290B
+#include "dsi_ota7290b.h"
+static struct panel_desc_s panel_desc = {
+    .panel_name = "OTA7290B-320x1280",
+    .dev_cfg = &dev_cfg_ota7290b_320x1280,
+    .hs_timing_cfg = &hs_timing_cfg_ota7290b_320x1280,
+    .dsi_init_cmds = dsi_init_cmds_ota7290b_320x1280,
+    .dsi_init_cmds_size = ARRAY_SIZE(dsi_init_cmds_ota7290b_320x1280)};
+#elif CONFIG_PANEL_OTA7290B_1920
+#include "dsi_ota7290b_1920.h"
+static struct panel_desc_s panel_desc = {
+    .panel_name = "OTA7290B-440x1920",
+    .dev_cfg = &dev_cfg_ota7290b_440x1920,
+    .hs_timing_cfg = &hs_timing_cfg_ota7290b_440x1920,
+    .dsi_init_cmds = dsi_init_cmds_ota7290b_440x1920,
+    .dsi_init_cmds_size = ARRAY_SIZE(dsi_init_cmds_ota7290b_440x1920)};
+#elif CONFIG_PANEL_ICN9707
+#include "dsi_icn9707.h"
+static struct panel_desc_s panel_desc = {
+    .panel_name = "ICN9707-480x1920",
+    .dev_cfg = &dev_cfg_icn9707_480x1920,
+    .hs_timing_cfg = &hs_timing_cfg_icn9707_480x1920,
+    .dsi_init_cmds = dsi_init_cmds_icn9707_480x1920,
+    .dsi_init_cmds_size = ARRAY_SIZE(dsi_init_cmds_icn9707_480x1920)};
+#elif CONFIG_PANEL_3AML069LP01G
+#include "dsi_3aml069lp01g.h"
+static struct panel_desc_s panel_desc = {
+    .panel_name = "3AML069LP01G-600x1024",
+    .dev_cfg = &dev_cfg_3AML069LP01G_600x1024,
+    .hs_timing_cfg = &hs_timing_cfg_3AML069LP01G_600x1024,
+    .dsi_init_cmds = dsi_init_cmds_3AML069LP01G_600x1024,
+    .dsi_init_cmds_size = ARRAY_SIZE(dsi_init_cmds_3AML069LP01G_600x1024)};
+#elif CONFIG_PANEL_ST7701
+#include "dsi_st7701.h"
+static struct panel_desc_s panel_desc = {
+    .panel_name = "ST7701-480x800",
+    .dev_cfg = &dev_cfg_st7701_480x800,
+    .hs_timing_cfg = &hs_timing_cfg_st7701_480x800,
+    .dsi_init_cmds = dsi_init_cmds_st7701_480x800,
+    .dsi_init_cmds_size = ARRAY_SIZE(dsi_init_cmds_st7701_480x800)};
+#elif CONFIG_PANEL_HW_MCU_ST7789V3
+#include "hw_mcu_st7789v3.h"
+static struct panel_desc_s panel_desc = {
+    .panel_name = "ST7789V3-240x320",
+    .stVoPubAttr =
+        {
+            .enIntfType = VO_INTF_HW_MCU,
+            .enIntfSync = VO_OUTPUT_USER,
+            .stSyncInfo = {.bSynm = 1,
+                           .bIop = 1,
+                           .u16FrameRate = 60,
+                           .u16Vact = 320,
+                           .u16Vbb = 5,
+                           .u16Vfb = 20,
+                           .u16Hact = 240,
+                           .u16Hbb = 20,
+                           .u16Hfb = 10,
+                           .u16Vpw = 5,
+                           .u16Hpw = 5,
+                           .bIdv = 0,
+                           .bIhs = 1,
+                           .bIvs = 1},
+            .stMcuCfg = st7789v3Cfg,
+        },
+};
+#else
+#include "dsi_hx8394_evb.h"
+static struct panel_desc_s panel_desc = {
+    .panel_name = "HX8394-720x1280",
+    .dev_cfg = &dev_cfg_hx8394_720x1280,
+    .hs_timing_cfg = &hs_timing_cfg_hx8394_720x1280,
+    .dsi_init_cmds = dsi_init_cmds_hx8394_720x1280,
+    .dsi_init_cmds_size = ARRAY_SIZE(dsi_init_cmds_hx8394_720x1280)};
+#endif
+
+#endif
diff --git a/components/cvi_mmf_sdk/cvi_panel/dsi_panels.h b/components/cvi_mmf_sdk/cvi_panel/dsi_panels.h
deleted file mode 100644
index 06ce4ab1..00000000
--- a/components/cvi_mmf_sdk/cvi_panel/dsi_panels.h
+++ /dev/null
@@ -1,93 +0,0 @@
-#ifndef _DSI_PANELS_
-#define _DSI_PANELS_
-
-#include "cvi_type.h"
-#include "cvi_mipi_tx.h"
-#include "cvi_comm_mipi_tx.h"
-
-#ifdef __cplusplus
-#if __cplusplus
-extern "C" {
-#endif
-#endif /* End of #ifdef __cplusplus */
-
-struct panel_desc_s {
-	char *panel_name;
-	struct combo_dev_cfg_s *dev_cfg;
-	const struct hs_settle_s *hs_timing_cfg;
-	const struct dsc_instr *dsi_init_cmds;
-	int dsi_init_cmds_size;
-};
-
-#if CONFIG_PANEL_ILI9488
-#include "dsi_ili9488.h"
-static struct panel_desc_s panel_desc = {
-	.panel_name = "ILI9488-320x480",
-	.dev_cfg = &dev_cfg_ili9488_320x480,
-	.hs_timing_cfg = &hs_timing_cfg_ili9488_320x480,
-	.dsi_init_cmds = dsi_init_cmds_ili9488_320x480,
-	.dsi_init_cmds_size = ARRAY_SIZE(dsi_init_cmds_ili9488_320x480)
-};
-#elif CONFIG_PANEL_ST7701S
-#include "dsi_st7701.h"
-static struct panel_desc_s panel_desc = {
-	.panel_name = "ST7701-480x800",
-	.dev_cfg = &dev_cfg_st7701_480x800,
-	.hs_timing_cfg = &hs_timing_cfg_st7701_480x800,
-	.dsi_init_cmds = dsi_init_cmds_st7701_480x800,
-	.dsi_init_cmds_size = ARRAY_SIZE(dsi_init_cmds_st7701_480x800)
-};
-#elif CONFIG_PANEL_ST7701_480x640
-#include "dsi_st7701_480x640.h"
-static struct panel_desc_s panel_desc = {
-	.panel_name = "ST7701-480x640",
-	.dev_cfg = &dev_cfg_st7701_480x640,
-	.hs_timing_cfg = &hs_timing_cfg_st7701_480x640,
-	.dsi_init_cmds = dsi_init_cmds_st7701_480x640,
-	.dsi_init_cmds_size = ARRAY_SIZE(dsi_init_cmds_st7701_480x640)
-};
-#elif CONFIG_PANEL_ST7703
-#include "dsi_st7703.h"
-static struct panel_desc_s panel_desc = {
-	.panel_name = "ST7703-640x480",
-	.dev_cfg = &dev_cfg_st7703_640x480,
-	.hs_timing_cfg = &hs_timing_cfg_st7703_640x480,
-	.dsi_init_cmds = dsi_init_cmds_st7703_640x480,
-	.dsi_init_cmds_size = ARRAY_SIZE(dsi_init_cmds_st7703_640x480)
-};
-#elif CONFIG_PANEL_HX8394
-#include "dsi_hx8394_evb.h"
-static struct panel_desc_s panel_desc = {
-	.panel_name = "HX8394-720x1280",
-	.dev_cfg = &dev_cfg_hx8394_720x1280,
-	.hs_timing_cfg = &hs_timing_cfg_hx8394_720x1280,
-	.dsi_init_cmds = dsi_init_cmds_hx8394_720x1280,
-	.dsi_init_cmds_size = ARRAY_SIZE(dsi_init_cmds_hx8394_720x1280)
-};
-#elif CONFIG_PANEL_OTA7290B
-#include "dsi_ota7290b_1920.h"
-static struct panel_desc_s panel_desc = {
-	.panel_name = "OTA7290B-440x1920",
-	.dev_cfg = &dev_cfg_ota7290b_440x1920,
-	.hs_timing_cfg = &hs_timing_cfg_ota7290b_440x1920,
-	.dsi_init_cmds = dsi_init_cmds_ota7290b_440x1920,
-	.dsi_init_cmds_size = ARRAY_SIZE(dsi_init_cmds_ota7290b_440x1920)
-};
-#else
-#include "dsi_hx8394_evb.h"
-static struct panel_desc_s panel_desc = {
-	.panel_name = "HX8394-720x1280",
-	.dev_cfg = &dev_cfg_hx8394_720x1280,
-	.hs_timing_cfg = &hs_timing_cfg_hx8394_720x1280,
-	.dsi_init_cmds = dsi_init_cmds_hx8394_720x1280,
-	.dsi_init_cmds_size = ARRAY_SIZE(dsi_init_cmds_hx8394_720x1280)
-};
-#endif
-
-#ifdef __cplusplus
-#if __cplusplus
-}
-#endif
-#endif /* End of #ifdef __cplusplus */
-
-#endif // _DSI_PANELS_
diff --git a/components/cvi_mmf_sdk/cvi_panel/hw_mcu_st7789v3.h b/components/cvi_mmf_sdk/cvi_panel/hw_mcu_st7789v3.h
new file mode 100644
index 00000000..a660b8eb
--- /dev/null
+++ b/components/cvi_mmf_sdk/cvi_panel/hw_mcu_st7789v3.h
@@ -0,0 +1,124 @@
+#ifndef _MCU_PARAM_ST7789V_H_
+#define _MCU_PARAM_ST7789V_H_
+
+#include <cvi_comm_vo.h>
+
+#define PANEL_COMM 0
+#define PANEL_DATA 1
+
+const VO_HW_MCU_CFG_S
+    st7789v3Cfg =
+        {
+            .pins = {.pin_num = 11,
+                     .d_pins =
+                         {
+                             {VO_VIVO_D6, VO_MUX_MCU_DATA0},
+                             {VO_VIVO_D5, VO_MUX_MCU_DATA1},
+                             {VO_VIVO_D4, VO_MUX_MCU_DATA2},
+                             {VO_VIVO_D3, VO_MUX_MCU_DATA3},
+                             {VO_VIVO_D2, VO_MUX_MCU_DATA4},
+                             {VO_VIVO_D1, VO_MUX_MCU_DATA5},
+                             {VO_VIVO_D0, VO_MUX_MCU_DATA6},
+                             {VO_VIVO_D7, VO_MUX_MCU_DATA7},
+                             {VO_MIPI_TXM1, VO_MUX_MCU_RD},
+                             {VO_MIPI_RXP5, VO_MUX_MCU_WR},
+                             {VO_MIPI_TXP1, VO_MUX_MCU_RS},
+                         }},
+            .mode = VO_MCU_MODE_RGB565,
+            .instrs =
+                {.instr_num = 74,
+                 .instr_cmd =
+                     {
+                         {.delay = 120, .data_type = PANEL_COMM, .data = 0x11},
+                         {.delay = 0, .data_type = PANEL_COMM, .data = 0x36},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x00},
+                         {.delay = 0, .data_type = PANEL_COMM, .data = 0x3A},
+                         {.delay = 0,
+                          .data_type = PANEL_DATA,
+                          .data = 0x05}, // 0x101 = 16 bit/pixel, 0x110 =
+                                         // 18bit/pixel
+                         {.delay = 0, .data_type = PANEL_COMM, .data = 0xB1},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x00},
+                         {.delay = 0, .data_type = PANEL_COMM, .data = 0xB2},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x0C},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x0C},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x00},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x33},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x33},
+                         {.delay = 0, .data_type = PANEL_COMM, .data = 0xB7},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x75},
+                         {.delay = 0, .data_type = PANEL_COMM, .data = 0xBB},
+                         {.delay = 0,
+                          .data_type = PANEL_DATA,
+                          .data = 0x1f}, // Vcom 0x1F: 0.875, 0x31: 1.35V
+                         {.delay = 0, .data_type = PANEL_COMM, .data = 0xC0},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x2C},
+                         {.delay = 0, .data_type = PANEL_COMM, .data = 0xC2},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x01},
+                         {.delay = 0, .data_type = PANEL_COMM, .data = 0xC3},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x13},
+                         {.delay = 0, .data_type = PANEL_COMM, .data = 0xC4},
+                         {.delay = 0,
+                          .data_type = PANEL_DATA,
+                          .data = 0x20}, // VDV, 0x20:0v
+                         {.delay = 0, .data_type = PANEL_COMM, .data = 0xC6},
+                         {.delay = 0,
+                          .data_type = PANEL_DATA,
+                          .data = 0x0F}, // Frame Rate control 0F: 60hz
+                         {.delay = 0, .data_type = PANEL_COMM, .data = 0xD0},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0xA4},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0xA1},
+                         {.delay = 0, .data_type = PANEL_COMM, .data = 0xD6},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0xA1},
+                         {.delay = 0, .data_type = PANEL_COMM, .data = 0xE0},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0xD0},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x0D},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x16},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x14},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x14},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x2F},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x38},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x54},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x48},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x0A},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x1C},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x19},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x1A},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x1D},
+                         {.delay = 0, .data_type = PANEL_COMM, .data = 0xE1},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0xD0},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x04},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x0C},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x08},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x09},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x23},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x39},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x54},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x48},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x30},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x14},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x14},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x19},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x1C},
+                         {.delay = 0, .data_type = PANEL_COMM, .data = 0x2A},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x00},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x00},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x00},
+                         {.delay = 0,
+                          .data_type = PANEL_DATA,
+                          .data = 0xEF}, // 240
+
+                         {.delay = 0, .data_type = PANEL_COMM, .data = 0x2B},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x00},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x00},
+                         {.delay = 0, .data_type = PANEL_DATA, .data = 0x01},
+                         {.delay = 0,
+                          .data_type = PANEL_DATA,
+                          .data = 0x40}, // 320
+
+                         {.delay = 0, .data_type = PANEL_COMM, .data = 0x29},
+                         {.delay = 0, .data_type = PANEL_COMM, .data = 0x2C},
+                     }},
+};
+
+#endif // _MCU_PARAM_ST7789V_H_
diff --git a/components/cvi_platform/media/src/media_video.c b/components/cvi_platform/media/src/media_video.c
index 51b578bd..846afad6 100644
--- a/components/cvi_platform/media/src/media_video.c
+++ b/components/cvi_platform/media/src/media_video.c
@@ -1,36 +1,36 @@
-#include <stdio.h>
-#include <unistd.h>
-#include <string.h>
-#include <errno.h>
-#include <aos/cli.h>
 #include "media_video.h"
-#include "rtos_types.h"
-#include "cvi_common.h"
+#include "cvi_buffer.h"
+#include "cvi_comm_cif.h"
 #include "cvi_comm_sys.h"
 #include "cvi_comm_vi.h"
-#include "sensor_cfg.h"
-#include "cvi_type.h"
+#include "cvi_common.h"
+#include "cvi_isp.h"
+#include "cvi_math.h"
+#include "cvi_mipi.h"
+#include "cvi_mipi_tx.h"
 #include "cvi_param.h"
-#include "cvi_vb.h"
+#include "cvi_sensor.h"
+#include "cvi_sns_ctrl.h"
 #include "cvi_sys.h"
+#include "cvi_type.h"
+#include "cvi_vb.h"
 #include "cvi_vi.h"
 #include "cvi_vpss.h"
-#include "cvi_isp.h"
-#include "cvi_sensor.h"
-#include "cvi_mipi.h"
-#include "cvi_buffer.h"
-#include "cvi_math.h"
-#include "cvi_comm_cif.h"
-#include "cvi_sns_ctrl.h"
-#include "cvi_mipi_tx.h"
+#include "rtos_types.h"
+#include "sensor_cfg.h"
+#include <aos/cli.h>
+#include <errno.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
 #if (!defined(CONFIG_SUPPORT_VO) || (CONFIG_SUPPORT_VO))
-#include "dsi_panels.h"
+#include "cvi_panels.h"
 #include "media_logo.h"
 #endif
 #include "cvi_gdc.h"
-#include "cvi_vo.h"
-#include "cvi_venc.h"
 #include "cvi_region.h"
+#include "cvi_venc.h"
+#include "cvi_vo.h"
 
 #include "cvi_ae_comm.h"
 #include "cvi_awb_comm.h"
@@ -962,47 +962,76 @@ static void _MEDIA_VIDEO_MIPI_TX_Init(void)
     printf("Init for MIPI-Driver-%s\n", desc->panel_name);
 }
 
-int MEDIA_VIDEO_VoInit(PARAM_VO_CFG_S * pstVoCtx)
-{
-    //vo init
-    CVI_S32 i = 0;
-    CVI_S32 j = 0;
-    PARAM_VO_CONFIG_S * pstVoConfig = NULL;
-    VO_VIDEO_LAYER_ATTR_S * pstLayerAttr = NULL;
-    VO_CHN_ATTR_S * pstVOChnAttr = NULL;
+static void _MEDIA_VIDEO_HW_MCU_Init(VO_PUB_ATTR_S *pstPubAttr) {
+  struct panel_desc_s *desc = &panel_desc;
 
-    if(pstVoCtx == NULL) {
-        return CVI_FAILURE;
-    }
+  pstPubAttr->enIntfSync = desc->stVoPubAttr.enIntfSync;
+  pstPubAttr->enIntfType = desc->stVoPubAttr.enIntfType;
+  pstPubAttr->stSyncInfo = desc->stVoPubAttr.stSyncInfo;
+  pstPubAttr->stMcuCfg = desc->stVoPubAttr.stMcuCfg;
 
-    if (pstVoCtx->u8VoCnt == 0) {
-        return CVI_SUCCESS;
-    }
-    _MEDIA_VIDEO_MIPI_TX_Init();
+  printf("Init for Driver-%s\n", desc->panel_name);
+}
 
-    for(i = 0 ;i < pstVoCtx->u8VoCnt;i++) {
-        pstVoConfig = &pstVoCtx->pstVoCfg[i].stVoConfig;
-        pstLayerAttr = &pstVoCtx->pstVoCfg[i].stLayerAttr;
-        MEDIA_CHECK_RET(CVI_VO_SetPubAttr(pstVoConfig->VoDev, &pstVoConfig->stVoPubAttr),"CVI_VO_SetPubAttr failed!\n");
-        MEDIA_CHECK_RET(CVI_VO_Enable(pstVoConfig->VoDev), "CVI_VO_Enable failed!\n");
-        MEDIA_CHECK_RET(CVI_VO_SetDisplayBufLen(pstVoCtx->pstVoCfg[i].VoLayer, pstVoConfig->u32DisBufLen),"CVI_VO_SetDisplayBufLen failed!\n");
-        MEDIA_CHECK_RET(CVI_VO_SetVideoLayerAttr(pstVoCtx->pstVoCfg[i].VoLayer, pstLayerAttr),
-            "CVI_VO_SetVideoLayerAttr failed!\n");
-        MEDIA_CHECK_RET(CVI_VO_EnableVideoLayer(pstVoCtx->pstVoCfg[i].VoLayer),
-            "CVI_VO_EnableVideoLayer failed!\n");
-        for(j = 0; j < pstVoCtx->pstVoCfg[i].u8ChnCnt; j++) {
-            pstVOChnAttr = &pstVoCtx->pstVoCfg[i].pstVOChnAttr[j];
-            MEDIA_CHECK_RET(CVI_VO_SetChnAttr(pstVoCtx->pstVoCfg[i].VoLayer, j, pstVOChnAttr),
-            "CVI_VO_SetChnAttr failed!\n");
-            CVI_VO_SetChnRotation(pstVoCtx->pstVoCfg[i].VoLayer,j,pstVoCtx->pstVoCfg[i].u8ChnRotation);
-            MEDIA_CHECK_RET(CVI_VO_EnableChn(pstVoCtx->pstVoCfg[i].VoLayer, j), "CVI_VO_EnableChn failed!\n");
-        }
-        if(pstVoCtx->pstVoCfg[i].u8Bindmode == true){
-            MEDIA_CHECK_RET(CVI_SYS_Bind(&pstVoCtx->pstVoCfg[i].stSrcChn, &pstVoCtx->pstVoCfg[i].stDestChn), "CVI_SYS_Bind(VPSS-VO)");
-        }
-    }
+int MEDIA_VIDEO_VoInit(PARAM_VO_CFG_S *pstVoCtx) {
+  // vo init
+  CVI_S32 i = 0;
+  CVI_S32 j = 0;
+  PARAM_VO_CONFIG_S *pstVoConfig = NULL;
+  VO_PUB_ATTR_S *pstPubAttr = NULL;
+  VO_VIDEO_LAYER_ATTR_S *pstLayerAttr = NULL;
+  VO_CHN_ATTR_S *pstVOChnAttr = NULL;
+
+  if (pstVoCtx == NULL) {
+    return CVI_FAILURE;
+  }
 
-    MEDIABUG_PRINTF("******start vo******\n");
+  if (pstVoCtx->u8VoCnt == 0) {
+    return CVI_SUCCESS;
+  }
+
+  pstPubAttr = &pstVoCtx->pstVoCfg[0].stVoConfig.stVoPubAttr;
+  printf("VoPubAttr.enIntfType: %#x\n", pstPubAttr->enIntfType);
+  if (pstPubAttr->enIntfType & VO_INTF_MIPI) {
+    _MEDIA_VIDEO_MIPI_TX_Init();
+  } else if (pstPubAttr->enIntfType & VO_INTF_HW_MCU) {
+    _MEDIA_VIDEO_HW_MCU_Init(pstPubAttr);
+  }
+
+  for (i = 0; i < pstVoCtx->u8VoCnt; i++) {
+    pstVoConfig = &pstVoCtx->pstVoCfg[i].stVoConfig;
+    pstLayerAttr = &pstVoCtx->pstVoCfg[i].stLayerAttr;
+    MEDIA_CHECK_RET(
+        CVI_VO_SetPubAttr(pstVoConfig->VoDev, &pstVoConfig->stVoPubAttr),
+        "CVI_VO_SetPubAttr failed!\n");
+    MEDIA_CHECK_RET(CVI_VO_Enable(pstVoConfig->VoDev),
+                    "CVI_VO_Enable failed!\n");
+    MEDIA_CHECK_RET(CVI_VO_SetDisplayBufLen(pstVoCtx->pstVoCfg[i].VoLayer,
+                                            pstVoConfig->u32DisBufLen),
+                    "CVI_VO_SetDisplayBufLen failed!\n");
+    MEDIA_CHECK_RET(
+        CVI_VO_SetVideoLayerAttr(pstVoCtx->pstVoCfg[i].VoLayer, pstLayerAttr),
+        "CVI_VO_SetVideoLayerAttr failed!\n");
+    MEDIA_CHECK_RET(CVI_VO_EnableVideoLayer(pstVoCtx->pstVoCfg[i].VoLayer),
+                    "CVI_VO_EnableVideoLayer failed!\n");
+    for (j = 0; j < pstVoCtx->pstVoCfg[i].u8ChnCnt; j++) {
+      pstVOChnAttr = &pstVoCtx->pstVoCfg[i].pstVOChnAttr[j];
+      MEDIA_CHECK_RET(
+          CVI_VO_SetChnAttr(pstVoCtx->pstVoCfg[i].VoLayer, j, pstVOChnAttr),
+          "CVI_VO_SetChnAttr failed!\n");
+      CVI_VO_SetChnRotation(pstVoCtx->pstVoCfg[i].VoLayer, j,
+                            pstVoCtx->pstVoCfg[i].u8ChnRotation);
+      MEDIA_CHECK_RET(CVI_VO_EnableChn(pstVoCtx->pstVoCfg[i].VoLayer, j),
+                      "CVI_VO_EnableChn failed!\n");
+    }
+    if (pstVoCtx->pstVoCfg[i].u8Bindmode == true) {
+      MEDIA_CHECK_RET(CVI_SYS_Bind(&pstVoCtx->pstVoCfg[i].stSrcChn,
+                                   &pstVoCtx->pstVoCfg[i].stDestChn),
+                      "CVI_SYS_Bind(VPSS-VO)");
+    }
+  }
+
+  MEDIABUG_PRINTF("******start vo******\n");
 
 #if (!defined(CONFIG_ALIOSLOGO) || (CONFIG_ALIOSLOGO))
 	//show logo
-- 
2.25.1

